### PR TITLE
Clean up git ignore and remove editor settings

### PR DESCRIPTION
**Pull request by blink1073**
Based on the `.gitignore` of the classic notebook.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,131 +1,23 @@
...
-
-# Runtime data
-pids
-*.pid
-*.seed
-
-# Dependency directory
-# https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
-node_modules
-npm-debug.log
```
Comment (vidartf) [inline .gitignore]: While npm-debug.log will no longer be put into cwd [as of 13 days ago](https://github.com/npm/npm/commit/04fca223a0f704b69340c5f81b26907238fad878), I would maybe keep this :)

Comment (blink1073) [inline .gitignore]: I'd say it was a mistake to have it there, because we shouldn't be checking in code if npm doesn't cleanly run.

Comment (vidartf) [inline .gitignore]: As long as it was a conscious choice :)

Comment (blink1073) [inline .gitignore]: Yep, thanks for keeping me honest ;)

**Pull request merged**